AWP-1641: Modal component

global-usman opened the pull request:
[AWP-1641](https://team-1637335847501.atlassian.net/browse/AWP-1641)
- Created modal component (header and footer) using useSlot hook
- Added test cases
- Created stories
- Added react portal to modal component
- Created ModalExample file




![image](https://github.com/Gloabal-Regulatory-Writing-Consulting/gxt-components/assets/113940169/9a1763ea-a3ef-43cf-a644-a9d215bdeda6)


[AWP-1641]: https://team-1637335847501.atlassian.net/browse/AWP-1641?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

global-usman committed:
AWP-1641/Modal component
- created modal component (header and footer) using useSlot hook
- added test cases
- created stories
- added react portal to modal component
- created ModalExample file

zamanaf commented on src/components/modal/__docs__/Modal.mdx:
````diff
@@ -0,0 +1,70 @@
+```
+
+#### Arguments
+- **ModalHeaderHeading** _`div`_ - The heading text for the modal header. It typically displays the main title of the modal.
````
can we use <Controls /> here from @storybook/blocks

zamanaf commented on src/components/modal/Modal.tsx:
```diff
@@ -0,0 +1,52 @@
+  children,
+  overlayBackground = "var(--neutral, #00000099)",
+  showBackground = true,
+  maxWidth = "288px",
```
can we use rem ?

zamanaf commented on src/components/modal/Modal.styles.ts:
```diff
@@ -0,0 +1,77 @@
+  margin: 0;
+`;
+
+export const ModalHeaderBody = styled.div<{ styles?: CSSProperties }>`
```
I think the style prop is unnecessary

zamanaf commented on src/components/modal/Modal.tsx:
```diff
@@ -0,0 +1,52 @@
+}) => {
+  return (
+    <ReactPortal wrapperId="react-portal-modal-container">
+      <StyledModalOverlay
```
can we close the modal onClicking on the overlay

zamanaf reviewed (approved):
LGTM

global-usman committed:
AWP-1641: Modal component

- Ensure modal close upon clicking the overlay
- Used @storybook/blocks
- Removed style prop
- Add isOpen prop to modal
- Update test cases
